bciers/libs/components/src/error/ErrorBox.tsx

```diff
@@ -0,0 +1,51 @@
+"use client";
+import { useEffect, useState } from "react";
+import * as Sentry from "@sentry/nextjs";
+import { ghgRegulatorEmail } from "@bciers/utils/src/urls";
+
+import { Alert, AlertTitle } from "@mui/material";
+
+interface Props {
+  error?: Error & { digest?: string };
+}
+export default function ErrorBox({ error }: Props) {
+  const [eventId, setEventId] = useState<string | null>(null);
+  useEffect(() => {
+    if (error) {
+      try {
+        // Attempt to log the error to Sentry
+        const id = Sentry.captureException(error);
+        setEventId(id); // Store Sentry event ID
+      } catch (sentryError) {
+        // If there's an error logging to Sentry, log it to the console
+        // eslint-disable-next-line no-console
+        console.error("Error logging to Sentry:", sentryError);
+      }
+    }
+  }, [error]);
+
+  return (
+    <div className="flex flex-col items-center text-bc-gov-links-color">
+      <h2>Something went wrong...</h2>
+      <Alert
+        severity="error"
+        sx={{
+          width: "100%",
+          maxWidth: "600px",
+          marginBottom: "1rem",
+        }}
+      >
+        <AlertTitle>
+          <strong>Error</strong>
+        </AlertTitle>
+        <p>An internal server error has occured.</p>
+        <p>
+          Please refresh the page. If the problem persists, contact{" "}
+          <a href={ghgRegulatorEmail}>ghgregulator@gov.bc.ca</a> for help and
+          include the reference code:{" "}
+          <strong>{eventId ? eventId : "pending"}</strong>
```

Comment:
I put 'pending' because it takes a moment for the code to show up, and if there's a problem logging to sentry, we'll never get a code